src/Dom/Document.php

```diff
@@ -219,6 +217,15 @@ final class Document extends DOMDocument {
 	 */
 	private $mustache_tags_replaced = false;
 
+	/**
+	 * Whether we had secured a doctype that needs restoring or not.
+	 *
+	 * This is an int as it receives the $count from the preg_replace().
+	 *
+	 * @var int
+	 */
+	private $secured_doctype = 0;
```

Comment:
Does this need to be reset in the `reset` method?

Reply:
No, it mustn't be reset. It's a scalar value, so cloning is not an issue and it needs to stay in the state it was set.